src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -238,7 +244,16 @@ else if(privacyCommand.equals("make public")){
       conversationStore.updateConversation(conversation);
     }
     else if(purpose.equals("Adding users")){
-      //TODO parse json here
+
+      String[] userNameArray = stringToArray(request.getReader().readLine());
```

Comment:
Whether you'll take my suggestion above about using Gson for parsing, or you want to do parsing yourself, you should be very careful about handling input coming from the client, and at least add `try/catch` logic around it. Remember clients can send anything to the server, including malformed JSON, and the server should never crash on bad input, which is what'll happen here probably.

We should also sanitize usernames before adding them by both: cleaning up any HTML in them (to avoid SQL injection threats and the like), and making sure they are valid user names of actual users in the system, so we don't clutter the conversation's member list with incorrect names that might break later on NPEs when fetching.

Let me know if this doesn't make sense or if we should discuss more.